### PR TITLE
Add test for not emit windows too early with sparse events

### PR DESCRIPTION
**Pull request by olukas**
Test is added as nightly test since it takes more than 15seconds. 

This PR also adds sparse event test plan to left menu on site.

Checklist
- [x] Tags Set
- [x] Milestone Set
- [N/A] Any breaking changes are documented
- [N/A] New public APIs have `@Nonnull/@Nullable` annotations
- [N/A] New public APIs have `@since` tags in Javadoc
- [N/A] For code samples, code sample main readme is updated

### CODE REVIEW

**Comment by olukas** [comment]
run-nightly-tests

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder-nightly-tests` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-nightly-tests/10/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-nightly-tests/10/jet-oss-pr-builder-nightly-tests-10.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   JobSubmissionSlownessRegressionTest.regressionTestForPR1488:116 Job submission rate should not decrease. First rate: 156.6194673010924, second rate: 88.2050931362341
[ERROR] Errors: 
[ERROR]   MulticastDiscoveryTest.when_jetClientCreated_then_connectsToJetCluster:99 ? IllegalState
[ERROR]   OperationLossTest.when_connectionDroppedWithoutMemberLeaving_then_jobRestarts:173 ? Completion
[INFO] 
[ERROR] Tests run: 58, Failures: 1, Errors: 2, Skipped: 2
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/StreamSourceStageTest.java [hunk 1]
```diff
@@ -165,6 +167,29 @@ public void when_sparseItemsWithIngestionTimestamps_then_noExtraLatency() {
...
+         .withIngestionTimestamps()
+         .window(WindowDefinition.session(15_000))
+         .aggregate(AggregateOperations.counting())
+         .writeTo(Sinks.list(sinkList));
+
+        Job job = instance.newJob(p);
+        assertEquals(0, sinkList.size());
+        map.put(5, 5);
+
+        assertTrueAllTheTime(() -> assertEquals(0, sinkList.size()), 10);
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/StreamSourceStageTest.java:187]: This is time-sensitive. If it takes 5 secs to submit the job, this assert can fail.

We can instead start measuring before submitting the job and we can safely check that more than 15 seconds passed before the first event appeared.

```
long start = System.nanoTime();
newJob(p);
assertTrueEventually( /* one item in sinkList */);
assertTrue(System.nanoTime() - start > 15 seconds);
```

**Review by viliam-durina — APPROVED** [review]
Approving since the test is correct, but my suggestion is to make it time-insensitive.

**Pull request merged**